### PR TITLE
GCE: Add os parameters to MigOsInfo interface

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind cleanup

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/os_reserved.go [hunk 1]
```diff
@@ -16,13 +16,23 @@ limitations under the License.
...
+	// Arch return system architecture.
+	Arch() SystemArchitecture
+}
+
 // OsReservedCalculator calculates the OS reserved values.
 type OsReservedCalculator interface {
 	// CalculateKernelReserved computes how much memory OS kernel will reserve.
 	// NodeVersion parameter is optional. If empty string is passed a result calculated using default node version will be returned.
-	CalculateKernelReserved(physicalMemory int64, os OperatingSystem, osDistribution OperatingSystemDistribution, arch SystemArchitecture, nodeVersion string) int64
+	CalculateKernelReserved(m MigOsInfo, physicalMemory int64, nodeVersion string) int64
```
Comment (BigDarkClown) [inline cluster-autoscaler/cloudprovider/gce/os_reserved.go:33]: I think now we can remove the nodeVersion as well, both from these functions and from Mig interface. It is not needed in the GCE implementation, and we are now able to add this in internal fork one using MigOsInfo.

**Review by BigDarkClown — COMMENTED** [review]
That change looks fantastic, it cleans up the interfaces so much! I added a little comment regarding removing the node version, but other than that - great job!

File: cluster-autoscaler/cloudprovider/gce/os_reserved.go (hunk 1 above)
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/cloudprovider/gce/os_reserved.go]: Right, done

**Review by BigDarkClown — APPROVED** [review]

**Pull request merged**